### ISSUE TITLE
Vue CLI integration

### ISSUE THREAD

**Issue by DavidRouyer** [DivanteLtd/vue-storefront#1974]
For now, the idea is to create a Vue CLI plugin which will automate the integration of Vue Storefront into an existing Vue CLI 3 project. (we'll think about nuxt.js later)
That way, we don't need to write a opinionated boilerplate (or multiple boilerplates), we can directly adapt to the user settings.

Steps to simplify the work:

- [ ] Replace [serviceworker-webpack-plugin with Workbox](https://github.com/DivanteLtd/vue-storefront/issues/1589) (Vue CLI uses Workbox in his PWA plugin)
- [x] Find a solution for not having to make a specific Webpack workflow for the service worker (https://github.com/DivanteLtd/vue-storefront/blob/master/core/service-worker/index.js) (Imports are the problem here)
- [ ] Implement SSR capabilities (https://github.com/Akryum/vue-cli-plugin-ssr)
- [ ] Remove scripts like [translation.preprocessor.js](https://github.com/DivanteLtd/vue-storefront/blob/master/core/i18n/scripts/translation.preprocessor.js) by replacing CSV files with JSON files for the translations
- [ ] ...

**Comment by filrak** [comment]
_Find a solution for not having to make a specific Webpack workflow for the service worker (https://github.com/DivanteLtd/vue-storefront/blob/master/core/service-worker/index.js) (Imports are the problem here)_
I believe vue-cli itself will handle the problem since their pwa plugin is generating service worker with based on workbox config. This config can accept external service worker src which will be bundled with the one generated with workbox.

So basically vue-cli gives us this separate build out of the box ;)

**Comment by pkarw** [comment]
We're all crossing fingers for this feature 🤝 🤞🤞

**Comment by filrak** [comment]
Ok i believe with mine and @NataliaTepluhina help we will be able to do it ;)

**Comment by fancydev18** [comment]
Hi @pkarw , why was this closed?
How can I help making it happen?

**Comment by pkarw** [comment]
We resigned from explicitly supporting Vue CLI - and @filrak started with the `create-vsf-app` kind of tool -> https://github.com/DivanteLtd/vue-storefront/pull/2856
